lib/pbench/server/api/resources/query_apis/datasets/datasets_contents.py

```diff
@@ -0,0 +1,216 @@
+                                    "type": "sym",
+                                    "linkpath": "sample1",
+                                }
+                            ],
+                            "run_data_parent": "ece030bdgfkjasdkf7435e6a7a6be804",
+                            "authorization": {"owner": "1", "access": "private"},
+                            "@timestamp": "2021-05-01T24:00:00",
+                        },
+                    },
+                    {
+                        "_index": "riya-pbench.v6.run-toc.2021-05",
+                        "_type": "_doc",
+                        "_id": "3bba25b62fhdgfajgsfdty6797ed06a",
+                        "_score": 0.0,
+                        "_source": {
+                            "parent": "/1-default",
+                            "directory": "/1-default/sample1",
+                            "mtime": "2021-05-01T24:00:00",
+                            "mode": "0o755",
+                            "name": "sample1",
+                            "ancestor_path_elements": ["1-default"],
+                            "files": [
+                                {
+                                    "name": "result.txt",
+                                    "mtime": "2021-05-01T24:00:00",
+                                    "size": 0,
+                                    "mode": "0o644",
+                                    "type": "reg",
+                                },
+                                {
+                                    "name": "user-benchmark.cmd",
+                                    "mtime": "2021-05-01T24:00:00",
+                                    "size": 114,
+                                    "mode": "0o755",
+                                    "type": "reg",
+                                },
+                            ],
+                            "run_data_parent": "ece030bdgfkjasdkf7435e6a7a6be804",
+                            "authorization": {"owner": "1", "access": "private"},
+                            "@timestamp": "2021-05-01T24:00:00",
+                        },
+                    },
+                ],
+            },
+        }
+
+        Output:
+            {
+                "directories":
+                [
+                    "sample1"
+                ],
+                "files": [
+                    {
+                        "name": "reference-result",
+                        "mtime": "2021-05-01T24:00:00",
+                        "size": 0,
+                        "mode": "0o777",
+                        "type": "sym",
+                        "linkpath": "sample1"
+                    }
+                ]
+            }
+        """
+        count = int(es_json["hits"]["total"]["value"])
+        if count == 0:
+            abort(
+                HTTPStatus.NOT_FOUND,
+                message=f"No directory '{context['parent']}' in '{context['run_id']}' contents.",
+            )
```

Comment:
@riya-17 : Sorry, my fault again ... it's been long enough that I've forgotten some of my own subclass hierarchy conventions, sadly. Doing an `abort` from the subclass will confuse the superclass `_call` logic. We need to report errors here by raising the `PostprocessError` ... basically we decided that we didn't want the indirect "support" code doing an `abort` directly. So replace the `abort` with the following code. In combination with the `expected_status` change, your tests will run.

```
        if count == 0:
            raise PostprocessError(
                HTTPStatus.NOT_FOUND, f"No directory '{context['parent']}' in '{context['run_id']}' contents."
            )
```

You'll also need to add the import:

```
from pbench.server.api.resources.query_apis import CONTEXT, PostprocessError
```

Be sure to manually run `black .` from the top level of your branch to be sure it'll be happy with these changes before you push a new commit. 😆

Reply:
yeah sure. Thank you

Reply:
I did see the usage of `PostprocessError` but I thought `abort` should work. I should have tried that, my bad.

Reply:
Yeah, the problem is that the Flask `abort` raises an exception, and the fall-back handler in `_call` in the base class will yell "ah ha! An unhandled exception!" and terminate with an internal server error. We could have caught that exception explicitly with a `pass`, but we decided to limit the use of `abort` instead.

However, given that it's clearly fairly easy to forget, maybe we should be catching the Flask exception anyway and issuing a specific message that'll be clearly visible in unit testing. 😆

Reply:
> We could have caught that exception explicitly with a pass
Or, perhaps, with a `raise`?

Given that the exception signifies a request from a nested scope to abort the request, I'm not sure that our fallback handler has anything to contribute....